### PR TITLE
PPML FGBoost add Client ID regulation to be consistent with NN

### PR DESCRIPTION
**Pull request by Litchilitchy**
Regulate the clientID when create `initFlContext()`

This is to make the behavior consistent with Pytorch & Keras NN

### CODE REVIEW

File: python/ppml/example/fgboost_regression/fgboost-tutorial.md [hunk 1]
```diff
@@ -2,17 +2,16 @@
 This example provides a step-by-step tutorial of running a FGBoost Regression task with 2 parties.
 ## 1. Key Concepts
 ### 1.1 FGBoost 
-**FGBoost** stands for Federated Gradient Boosted Tree algorithm. It allows multiple parties to run a federated gradient boosted decision tree application.
+**FGBoost** implements a Vertical Federated Learning algorithm for XGBoost. It allows multiple parties to run a federated gradient boosted decision tree application.
 ### 1.2 PSI
-**PSI** stands for Private Set Intersection algorithm. It compute the intersection of data from different parties and return the result so that parties know which data they could collaborate with.
+**PSI** stands for Private Set Intersection algorithm. It compute the intersection of data from different parties and return the intersection result (the data IDs which all the parties owns) so that parties know which data they could collaborate with.
```
Comment (jason-dai) [inline python/ppml/example/fgboost_regression/fgboost-tutorial.md:7]: the common data IDs that all the parties holds

File: python/ppml/example/fgboost_regression/fgboost-tutorial.md [hunk 2]
```diff
@@ -2,17 +2,16 @@
...
 ### 1.3 SGX
 **SGX** stands for [Software Guard Extensions](https://www.intel.com/content/www/us/en/developer/tools/software-guard-extensions/overview.html). It helps protect data in use via unique application isolation technology. The data of application running in SGX would be protected.
```
Comment (jason-dai) [inline python/ppml/example/fgboost_regression/fgboost-tutorial.md:9]: need to update

**Comment by qiyuangong** [comment]
This PR changes too many files. Can we split it into 2-3 PRs?

**Comment by Litchilitchy** [comment]
> This PR changes too many files. Can we split it into 2-3 PRs?

The files mainly contain the refactors and they all depend on each other. So I think they have to be together

**Pull request merged**